nexus/server.go

```diff
@@ -395,30 +395,44 @@ func (h *httpHandler) cancelOperation(service, operation, operationID string, wr
 	writer.WriteHeader(http.StatusAccepted)
 }
 
-// parseRequestTimeoutHeader checks if the Request-Timeout HTTP header is set and returns the parsed duration if so.
+// parseTimeoutHeader checks if the headerName HTTP header is set and returns the parsed duration if so.
 // Returns (0, true) if unset. Returns ({parsedDuration}, true) if set. If set and there is an error parsing the
 // duration, it writes a failure response and returns (0, false).
-func (h *httpHandler) parseRequestTimeoutHeader(writer http.ResponseWriter, request *http.Request) (time.Duration, bool) {
-	timeoutStr := request.Header.Get(HeaderRequestTimeout)
+func (h *httpHandler) parseTimeoutHeader(writer http.ResponseWriter, request *http.Request, headerName string) (time.Duration, bool) {
+	timeoutStr := request.Header.Get(headerName)
 	if timeoutStr != "" {
 		timeoutDuration, err := parseDuration(timeoutStr)
 		if err != nil {
 			h.logger.Warn("invalid request timeout header", "timeout", timeoutStr)
-			h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeBadRequest, "invalid request timeout header"))
+			h.writeFailure(writer, HandlerErrorf(HandlerErrorTypeBadRequest, "invalid timeout header"))
 			return 0, false
 		}
 		return timeoutDuration, true
 	}
 	return 0, true
 }
 
-// contextWithTimeoutFromHTTPRequest extracts the context from the HTTP request and applies the timeout indicated by
-// the Request-Timeout header, if set.
-func (h *httpHandler) contextWithTimeoutFromHTTPRequest(writer http.ResponseWriter, request *http.Request) (context.Context, context.CancelFunc, bool) {
-	requestTimeout, ok := h.parseRequestTimeoutHeader(writer, request)
+// contextWithTimeoutFromHTTPRequest extracts the context from the HTTP request and applies the minimum timeout of the
+// timeouts indicated by the Request-Timeout header and any additional specified headers, if set.
+func (h *httpHandler) contextWithTimeoutFromHTTPRequest(writer http.ResponseWriter, request *http.Request, headerNames ...string) (context.Context, context.CancelFunc, bool) {
+	requestTimeout, ok := h.parseTimeoutHeader(writer, request, HeaderRequestTimeout)
 	if !ok {
 		return nil, nil, false
 	}
```

Comment:
What if I only set an operation timeout? Could just:

```suggestion
	headerNames = append(headerNames, HeaderRequestTimeout)
```

(only acceptable to mutate the trailing param because this is a private function and you know caller isn't reusing header name slice)

Reply:
Only setting an operation timeout is valid. The `parseTimeoutHeader` will only return `!ok` if there is a parsing error. It will return `0, true` if unset.

About modifying the trailing param: I thought about a couple different options but didn't love any of them. I settled on this because I thought it made it more explicit that `Request-Timeout` is always checked. But I am not committed to this approach if others feel strongly.

Reply:
You're good, I misunderstood what absence of a request timeout did here. No need to change anything. (may be worth a test or two though)